src/library_syscall.js

```diff
@@ -119,8 +122,33 @@ var SyscallsLibrary = {
       if (suggest) FS.close(suggest);
       return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
     },
-    doReadv: function(stream, iov, iovcnt, offset) {
+#if EMTERPRETIFY_ASYNC
```

Comment:
the code duplication with the non-async version worries me. Coudl be just add the extra function parameter when async, but leave all the logic of reading `iov` etc. shared?